### PR TITLE
Various patches

### PR DESCRIPTION
**Pull request by BourgondAries**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -124,6 +124,9 @@ use std::boxed::Box;
 #[cfg(not(feature = "std"))]
 use alloc::boxed::Box;
 
+/// Error type that allows custom implementations of the Serialize trait.
+pub use ser::Error;
```
Comment (dpc) [inline src/lib.rs]: I'm not sure about this one. Serialization is a separate part, and if ser::Error` was to be in root namespace then I guess everything from `ser::*` should.

File: crates/scope/lib.rs [hunk 2]
```diff
@@ -101,10 +103,9 @@ impl Drop for ScopeGuard {
 pub fn logger() -> Logger {
     TL_SCOPES.with(|s| {
         let s = s.borrow();
-        if s.is_empty() {
-            (*GLOBAL_LOGGER.get()).clone()
-        } else {
-            s[s.len() - 1].clone()
+        match s.last() {
+          Some(logger) => logger.clone(),
```
Comment (dpc) [inline crates/scope/lib.rs]: It's not a big deal, but let's stick with `rustfmt` mandated formatting (4 spaces etc.). I generally from time to time `rustfmt` everything, so not a big deal.

File: src/lib.rs (hunk 1 above)
Comment (BourgondAries) [inline src/lib.rs]: That's true but wouldn't it be better to then use `use slog::ser::*;` and move `Serialize` and `Record` into the `ser` module?

Comment (dpc) [inline src/lib.rs]: I'm not sure. Having it in separate module kind of helps keeping the less-often used bits from polutting the more important ones (eg. in documentation).

**Comment by dpc** [comment]
I have cherry-picked the patches that I found OK. I'm still undecided on the `ser::` module, so I'm just leaving it for now as is.

Thanks!

**Pull request closed without merging**